sdk/src/metrics/meter.cc

```diff
@@ -39,6 +40,13 @@ nostd::unique_ptr<metrics::Counter<uint64_t>> Meter::CreateUInt64Counter(
     nostd::string_view description,
     nostd::string_view unit) noexcept
 {
+  if (!ValidateInstrument(name, description, unit))
+  {
+    OTEL_INTERNAL_LOG_ERROR("Meter::CreateUInt64Counter - failed. Invalid parameters."
+                            << name << " " << description << " " << unit);
+    return nostd::unique_ptr<metrics::Counter<uint64_t>>(
+        new metrics::NoopCounter<uint64_t>(name, description, unit));
```

Comment:
The calling instrumented application will not notice the error if returning a noop counter.

Not crashing now with a noop is not helping.
Instead, bugs will be filed later against opentelemetry-cpp about instrument that produce no data.

How about returning a nullptr, forcing the calling application to do something about it ?

Reply:
Yes, good point. I too thought about it, but also felt that making some critical application crash for some issue in telemetry configuration is not a good idea.  We can also be more explicit in error message stating that measurements won't be recorded.

This is also inline with the general error handling [guidelines ](https://github.com/jack-berg/opentelemetry-specification/blob/6307e44b88dd8f654134946c765cd22dc01c8455/specification/error-handling.md#error-handling-and-performance) - 
``
Whenever API call returns values that is expected to be non-null value - in case of error in processing logic - SDK MUST return a "no-op" or any other "default" object that was (ideally) pre-allocated and readily available. This way API call sites will not crash on attempts to access methods and properties of a null objects.
``

Reply:
Thanks for the details. Ok with noop then.